modules/biobank/php/collection.class.inc

```diff
@@ -0,0 +1,107 @@
+<?php declare(strict_types=1);
+/**
+ * Specimen Object Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Specimen Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+namespace LORIS\biobank;
+
+/**
+ * Specimen Object Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Specimen Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class Collection extends Process
+{
+    /**
+     * Persistent Instance variables.
+     *
+     * These variables are directly mapped to the fields of the database schema
+     * that describe a specimen entity.
+     *
+     * @access private
+     */
+    private $quantity; //float
+    private $unitId;   //int
+
+    /**
+     * Sets the current quantity of specimen available. This associated unit
+     * can be found in the `biobank_unit` using the unit ID.
+     *
+     * @param float $quantity the current amount of specimen available
+     *
+     * @return void
+     */
+    public function setQuantity(float $quantity) : void
```

Comment:
It doesn't seem like float is a good idea for tracking quantities in a biobank. This seems like an area where you want a well defined fixed amount of precision, not rounding error.

(Also same comment about the getter/setters and return type as the class above.)

Reply:
What do you recommend as an alternative to using `float`?

Reply:
yes, otherwise you'd need to come up with some fixed width integer classes since PHP doesn't natively support them.. but `float` will lose precision and should only be used when the speed of doing math with them is more important than the accuracy, which I don't think is the case here.

Reply:
Sounds good. Done.